uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequestCalendarFeedConfig.php

```diff
@@ -232,4 +232,185 @@ public static function getValuesArray($id) {
       return [];
     }
   }
+
+  /**
+   * Implementation of the addSelectWhereClause:
+   * This ensures that the administrator (user with 'can administer calendar feeds')
+   * permission can access all feed configs while users without this permission can
+   * only access feed configs which they have visibility access to. i.e if they belong
+   * to any of the department or location visibility set  for a feed config.
+   */
+  public function addSelectWhereClause() {
+    if (CRM_Core_Permission::check([['can administer calendar feeds']])) {
+      return;
+    }
+
+    $limitedVisibilityFeedConfigs = $this->getLimitedVisibilityFeedConfigs();
+
+    if (empty($limitedVisibilityFeedConfigs)) {
+      return;
+    }
+
+    $accessibleDepartments = $this->getAccessibleDepartmentsForLoggedInContact();
```

Comment:
These methods are all really well done, but I wonder if maybe it would be cleaner to put this logic into a separate class, for example `$leaveRequestFeedConfigService->getInaccessibleConfigsForContact($contactID);` or 

```
$allConfigs = $leaveRequestFeedConfigService->findAll();
$accessibleConfigs = $leaveRequestFeedConfigService->findAllAccessibleToContact($contactId);
// do some diff of the IDs to get inaccessible ones
```

Reply:
I think it would be a bit too much to create a separate class for this when the methods are only used by the LeaveRequestCalendarFeedConfig. addSelectWhereClause method only now. But If we ever need to get available feed configs for a user anytime I agree it would be good to move to a class then.

Reply:
That's fine, I don't think there's a need to change it now. But in the future I would say it makes things easier to test and reduces class size if we encapsulate logic inside a class, _even if_ that class is only used in one place.

Reply:
Good point